x-pack/plugins/observability/public/pages/cases/use_fetch_alert_data.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useCallback, useMemo } from 'react';
+import { isEmpty } from 'lodash';
+
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { HttpSetup } from 'kibana/public';
+import { BASE_RAC_ALERTS_API_PATH } from '../../../../rule_registry/common/constants';
+import { useDataFetcher } from './use_data_fetcher';
+
+export const useFetchAlertData = (alertIds: string[]): [boolean, Record<string, unknown>] => {
+  const validIds = useMemo(() => getValidValues(alertIds), [alertIds]);
+  const shouldExecuteApiCall = useCallback((ids: string[]) => ids.length > 0, []);
+
+  const {
+    fetch,
+    cancel,
+    loading,
+    data: alerts,
+  } = useDataFetcher<string[], Record<string, unknown> | undefined>({
+    paramsForApiCall: validIds,
+    initialDataState: undefined,
+    executeApiCall: fetchAlerts,
+    shouldExecuteApiCall,
+  });
+
+  useEffect(() => {
+    fetch();
```

Comment:
this is a way to do it but it adds extra complications that are not necessary. 

You can start the fetch immediately inside `useDataFetcher` and invoke it here, have it return a promise and await for that promise here.